_posts/2025-05-16-Introduction_to_bidirectional-languages-in-your-website.md

```diff
@@ -0,0 +1,127 @@
+
+## Examples on what to flip when displaying RTL writing systems
+
+- Reading order
+- Directional icons e.g. arrow icons
+- Image locations
+- Buttons
+- Columns
+- Page number
+
+Some common mistakes are flipping elements that shouldn't be flipped such as images, original Arabic digits (1, 2, 3 etc.) and media player buttons.
+
+Next I will demonstrate what to do and consider when creating a website for both LTR and RTL layouts.
+
+## Tech stack
+
+- React
+- Typescript
+- Vite
+- TailwindCSS
+
+Below is a simple site on my two lovely cats Ninni and Eino because who doesn't love cats?
+
+![[simple website 1.png]](/img/2025-05-16-introduction-to-bidirectional-languages-in-your-website/1.png)
+
+I have made two localization files `en.json` and `fa.json`. I will handle the localization with [react-i18next](https://www.react.i18next.com/).
+
+Let's see what happens when I change the language without doing anything with RTL layout in mind.
+
+![[simple website 2.png]](/img/2025-05-16-introduction-to-bidirectional-languages-in-your-website/2.png)
+
+Not as bad as you would think but there are few things that are wrong here. The header should be flipped and the names `Ninni` and `Eino` beside the pictures should be aligned to the left side.
+
+Fortunately there is a quick fix for this. We can set the directionality for the element's using the `dir` attribute. We can set it on specific elements but for this case we want it on the whole html.
+
+```html
+<html dir="rtl"></html>
+```
+
+Let's change the directionality every time the language changes.
+
+```typescript
+useEffect(() => {
+  const dir = i18n.dir(i18n.language);
+  document.documentElement.dir = dir;
+}, [i18n, i18n.language]);
+```
+
+Let's see what's changed.
+
+![[simple website 3.png]](/img/2025-05-16-introduction-to-bidirectional-languages-in-your-website/3.png)
+
+Everything is now flipped correctly and we are finished here right? Well there seems to be few things still that are wrong here.
+
+### Margin and padding
+
+If we take a closer look we can see that the margins and paddings aren't flipped when the directionality changes.
+
+There are out-of-the-box solutions that deals with margins and padding etc. automatically such as [rtlcss](https://rtlcss.com/learn/) and [tailwindcss-rtl](https://www.npmjs.com/package/tailwindcss-rtl). For demonstration purposes I will show you how to deal with them manually.
+
+When creating a website with support for both directionalities you should avoid using `margin` and `padding`. Instead we should use `margin-inline-start`, `margin-inline-end`,
+`padding-inline-start` and `padding-inline-start` which aligns items depending on the directionality.
+
+[Tailwind](https://tailwindcss.com/blog/tailwindcss-v3) offers RTL and LTR modifiers which can be used.
```

Comment:
I would maybe add something like " for these use-cases" at the end of the sentence.